trio/_core/tests/test_windows.py

```diff
@@ -38,5 +41,109 @@
             print("end loop")
 
 
+async def test_WaitForMultipleObjects_sync():
+    # One handle
+    handle1 = kernel32.CreateEventA(ffi.NULL, True, False, ffi.NULL)
+    t = Thread(target=WaitForMultipleObjects_sync, args=(handle1,))
+    t.start()
+    kernel32.SetEvent(handle1)
+    t.join()  # the test succeeds if we do not block here :)
+    kernel32.CloseHandle(handle1)
```

Comment:
I don't think this test needs a thread at all – there's no particular reason to think that the thread is actually running before we call `SetEvent`, so you can capture the important part of the test by doing:

```python
handle1 = kernel32.CreateEventA(ffi.NULL, True, False, ffi.NULL)
kernel32.SetEvent(handle1)
WaitForMultipleObjects_sync(handle1)
kernel32.CloseHandle(handle1)
```

Reply:
> there's no particular reason to think that the thread is actually running before we call `SetEvent`

Oh dear, I overlooked that :)

Signaling before waiting is one option (and I added it for the fast tests), but I think it would be good to also test actually "acquiring the wait" and then setting the handle to "release the wait". Put that in a `@slow` test.

Reply:
Makes sense :-)